internal/supervisor/import.go

```diff
@@ -99,9 +105,96 @@ func NewImportSupervisor(awsClient toolsAWS.AWS, awat AWATClient, store installa
 	}
 }
 
+// completeImports transitions Installations back to stable after
+// checking with the AWAT to make sure that it has detected the
+// completion of the import
+func (s *ImportSupervisor) completeImports() error {
+	installationList, err := s.store.GetInstallations(
+		&model.InstallationFilter{
+			Paging: model.AllPagesNotDeleted(),
+			State:  model.InstallationStateImportComplete,
+		}, false, false)
+	if err != nil {
+		return errors.Wrap(err, "failed to list Installations")
+	}
+
+	for _, installation := range installationList {
+		importStatusList, err := s.awatClient.GetImportStatusesByInstallation(installation.ID)
+		if err != nil {
+			s.logger.WithError(err).Warnf("failed to get Import status for Installation %s", installation.ID)
+			continue
+		}
+		if len(importStatusList) == 0 {
+			s.logger.Warnf("no imports found for Installation with state %s, %s", installation.State, installation.ID)
+			continue
+		}
+		// find the most recently completed Import; that's the only one we
+		// care about
+		var mostRecentImport *awat.ImportStatus
+		for _, is := range importStatusList {
+			if is.State != awat.ImportStateComplete {
+				// An import is still running against this Installation.
+				// Multiple parallel imports against an Installation are not supported.
+				break
+			}
+			if mostRecentImport == nil || is.CompleteAt > mostRecentImport.CompleteAt {
+				mostRecentImport = is
+			}
+		}
+		if mostRecentImport == nil {
+			// an Import might still be running
+			continue
+		}
+
+		locked, err := s.store.LockInstallation(installation.ID, s.ID)
+		if err != nil {
+			return errors.Wrapf(err, "failed to lock Installation %s", installation.ID)
+		}
+		if !locked {
+			return errors.Errorf("failed to lock Installation %s", installation.ID)
+		}
+		defer func() {
+			unlocked, err := s.store.UnlockInstallation(installation.ID, s.ID, false)
+			if err != nil {
+				s.logger.WithError(err).Warnf("failed to unlock Installation %s to mark Import %s complete", installation.ID, mostRecentImport.ID)
+				return
+			}
+			if !unlocked {
+				s.logger.Warnf("failed to unlock Installation %s to mark Import %s complete", installation.ID, mostRecentImport.ID)
+			}
+		}()
+
+		// no Imports are running, the Installation may be moved to stable
+		installation.State = model.InstallationStateStable
+		err = s.store.UpdateInstallation(installation)
+		if err != nil {
+			s.logger.WithError(err).Error("failed to mark Installation stable")
+			// this continue is currently redundant but if anything is ever
+			// added to this function after this if stanza, it will be good
+			// to already have so it isn't omitted by mistake
```

Comment:
I do not think it is redundant because of webhook 🙂

Reply:
haha, indeed